topology/nhlt/intel/dmic-nhlt.c

```diff
@@ -0,0 +1,1002 @@
+						  &vertical_angle_begin, &vertical_angle_end,
+						  &horizontal_angle_begin, &horizontal_angle_end);
+
+			if (ret) {
+				fprintf(stderr, "nhlt_dmic_get_ep: dmic_get_mic_params failed\n");
+				return ret;
+			}
+
+			mic_conf.type = type;
+			mic_conf.panel = panel;
+			mic_conf.speaker_position_distance = speaker_position_distance;
+			mic_conf.horizontal_offset = horizontal_offset;
+			mic_conf.vertical_offset = vertical_offset;
+			mic_conf.frequency_low_band = frequency_low_band;
+			mic_conf.frequency_high_band = frequency_high_band;
+			mic_conf.direction_angle = direction_angle;
+			mic_conf.elevation_angle = elevation_angle;
+			mic_conf.vertical_angle_begin = vertical_angle_begin;
+			mic_conf.vertical_angle_end = vertical_angle_end;
+			mic_conf.horizontal_angle_begin = horizontal_angle_begin;
+			mic_conf.horizontal_angle_end = horizontal_angle_end;
+
+			memcpy(ep_target, &mic_conf, sizeof(struct mic_vendor_config));
+			ep_target += sizeof(struct mic_vendor_config);
+		}
+	} else {
+		memcpy(ep_target, &mic_s_conf, sizeof(struct mic_array_device_specific_config));
+		ep_target += sizeof(struct mic_array_device_specific_config);
+	}
+
+	if (extension) {
+		mic_ext.snr = snr;
+		mic_ext.sensitivity = sensitivity;
+		memcpy(ep_target, &mic_ext, sizeof(struct mic_snr_sensitivity_extension));
+		ep_target += sizeof(struct mic_snr_sensitivity_extension);
+	}
+
+	memcpy(ep_target, &f_conf, sizeof(struct formats_config));
+	ep_target += sizeof(struct formats_config);
+
+	memcpy(ep_target, &f_conf1, sizeof(struct format_config));
+	ep_target += sizeof(struct format_config);
+
+	ret = dmic_get_vendor_blob(tplg_pp, ep_target);
+	if (ret) {
+		fprintf(stderr, "nhlt_dmic_get_ep: dmic_get_vendor_blob failed\n");
+		return ret;
+	}
+
+	return 0;
+}
+
+/*
+ * Set dmic parameters from topology for dmic coefficient calculation.
+ *
+ * Coefficients are recalculated in case of multiple DAIs in topology and might affect each other.
+ *
+ * You can see an example of topology v2 config of dmic below. In this example the default
+ * object parameters are spelled out for clarity. General parameters like clk_min are parsed with
+ * set_dmic_data and pdm object data with set_pdm_data. Number of pdm's can vary from 1 to 2. Values
+ * are saved into intermediate structs and the vendor specific blob is calculated at the end of
+ * parsing with dmic_calculate.
+ *
+ *	DMIC."0" {
+ *		name NoCodec-6
+ *		id 6
+ *		index 0
+ *		driver_version		1
+ *		io_clk                  38400000
+ *		clk_min			500000
+ *		clk_max			4800000
+ *		duty_min		40
+ *		duty_max		60
+ *		sample_rate		48000
+ *		fifo_word_length	16
+ *		unmute_ramp_time_ms	200
+ *		num_pdm_active          2
+ *
+ *		# PDM controller config
+ *		Object.Base.pdm_config."0" {
+ *			ctrl_id	0
+ * 			mic_a_enable	1
+ *			mic_b_enable	1
+ *			polarity_a	0
+ *			polarity_b	0
+ *			clk_edge	0
+ *			skew		0
+ *		}
+ *      }
+ */
+int nhlt_dmic_set_params(struct tplg_pre_processor *tplg_pp,
```

Comment:
same here, its really get_params and to be more precise, we should call it nhlt_dmic_get_input_params()